### PR TITLE
add postgres monitoring, initial commit

### PR DESCRIPTION
**Pull request by suckatrash**
This commit adds the capability for collecting postgres metrics

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -105,6 +105,18 @@ By default, this will create a set of certificates in `/etc/grafana` that are ba
...
+
+To allow telegraf to connect to postgres, apply the included profile class to the postgres host:
+
+```
+class {'puppet_metrics_dashboard::profile::postgres':
+  pg_password => 'changeme',
+}
+```
+
+Additionally, in the main `puppet_metrics_dashborad` class, you'll need to pass values for `postgres_host` and 'postgres_password` (must match value of `pg_password` above)
```
Comment (jarretlavallee) [inline README.md]: If these parameters should be the same, can we set the default in the profile to be the init class?

File: manifests/params.pp [hunk 2]
```diff
@@ -28,6 +28,8 @@
   $configure_telegraf     =  true
   $master_list            =  [$::settings::certname]
   $puppetdb_list          =  [$::settings::certname]
+  $postgres_host          =  undef
```
Comment (jarretlavallee) [inline manifests/params.pp]: Would we want this to default to settings::certname as well? The default value shouldn't be used unless the profile is included right?

File: manifests/profile/postgres.pp [hunk 3]
```diff
@@ -0,0 +1,23 @@
+class puppet_metrics_dashboard::profile::postgres (
+  $pg_password
+){
+
+  pe_postgresql::server::pg_hba_rule{ 'allow telegraf user':
```
Comment (jarretlavallee) [inline manifests/profile/postgres.pp]: pe_postgresql breaks using this on open source puppet. I am not sure if we tested it with open source, but maybe we can add the puppetlabs-postgresql module instead? I am not sure if that module provides the same capabilities.

File: manifests/profile/postgres.pp [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    type        => 'hostnossl',
+    database    => 'pe-puppetdb',
+    user        => 'telegraf',
+    auth_method => 'password',
+    address     => '192.168.0.8/32',
```
Comment (jarretlavallee) [inline manifests/profile/postgres.pp]: Will this be `postgres_host` after the testing phase is done?

File: manifests/profile/postgres.pp [hunk 5]
```diff
@@ -0,0 +1,23 @@
+class puppet_metrics_dashboard::profile::postgres (
+  $pg_password
```
Comment (jarretlavallee) [inline manifests/profile/postgres.pp]: Would this be a good candidate for Sensitive? Can we enable Cert based auth, or is that too much of a reach for this initial development?

File: README.md (hunk 1 above)
Comment (suckatrash) [reply README.md]: good idea!

File: manifests/params.pp (hunk 2 above)
Comment (suckatrash) [reply manifests/params.pp]: yeah, I think this would work

File: manifests/profile/postgres.pp (hunk 4 above)
Comment (jarretlavallee) [reply manifests/profile/postgres.pp]: Oh that makes more sense now. How about a puppetdb_query? We could look for the main class and pull the fact from the node or depend on the certname to be resolvable.

**Comment by suckatrash** [comment]
Scrapping this, will submit a new PR

**Pull request closed without merging**